### ISSUE TITLE
runPCA: svd() function - infinite or missing values in 'x'

### ISSUE THREAD

**Issue by d-kolbe** [rounakdey/FastSparseGRM#5]
Hi there!
I am trying to analyse a WES dataset with the STAARpipeline and as part of that using FastSparseGRM.
First of all, thank you for providing this package.

I came across an issue however, which may very well be an issue with my data.

When running Step4: runPCA, I get the following error:
svd() function - infinite or missing values in 'x' (something similar)

For the code:
init<-svd(H,nv=0)$u

I read this was caused by NA values in the matrix. I therefore replaced all NAs with 0 for both the "H" matrix and "T".
The script worked and I got my PCs, however, all my files are either all "0" now or "inf".

Could someone perhaps let me know where I could have gone wrong? (please let me know what data I should provide)

Thank you very much!

Daniel

**Comment by d-kolbe** [comment]
I just realised I didn't remove sex chromosomes, which is likely causing the issue. I will rerun and update if the issues persists/is solved. Sorry!

**Comment by rounakdey** [comment]
Hi, the problem is because of monomorphic SNPs in the BED file. Please filter the BED file to have only SNPs with MAC>=1, and the problem should be solved.

**Comment by d-kolbe** [comment]
Hi Rounak,
thank you for your fast response! That helped! In theory monomorphic sites should've been removed as I filtered by MAF, however there were some stragglers. I ended up getting a new error using the WES data - something along the lines of "exceeding buffer size".
I ended up using a different dataset with common variants only from a GSA chip, which as Xihao Li pointed out makes more sense anyway, and everything worked fine.
All the best,
Daniel
